python/cudf_polars/cudf_polars/dsl/translate.py

```diff
@@ -376,7 +376,13 @@ def _(node: pl_expr.Window, visitor: NodeTraverser, dtype: plc.DataType) -> expr
 
 @_translate_expr.register
 def _(node: pl_expr.Literal, visitor: NodeTraverser, dtype: plc.DataType) -> expr.Expr:
-    value = pa.scalar(node.value, type=plc.interop.to_arrow(dtype))
+    if isinstance(node.value, PySeries):
+        value = node.value.to_arrow()
+        # TODO: casting from large string to string is not ideal
+        if value.type == pa.large_string():
+            value = value.cast(pa.string())
+    else:
+        value = pa.scalar(node.value, type=plc.interop.to_arrow(dtype))
```

Comment:
Instead of this, can you cherry-pick https://github.com/wence-/cudf/commit/a6b48e4108711a699e94349892099e2db5c2ff86